### DISCUSSION TITLE
Are changes in extension preferences propagated dynamically?

### DISCUSSION THREAD

**Discussion by hedgieinsocks** [Ulauncher/Ulauncher#1401] [Extensions and Themes]
Could you please clarify whether ulauncher 6 extensions dynamically receive changes in the preferences. 

I've created a simple script launcher [extension](https://github.com/hedgieinsocks/ulauncher-launchpad/blob/master/main.py) for myself that requires setting the script dir. 

And while developing, I noticed that I must disable/enable the extension to pick up the new saved value. 

I'm not a python dev, so I'm just asking whether it's supposed to be like that or I'm doing something wrong and I must subscribe to some settings event and manually update self.preferences?

**Comment by friday** [comment]
Sorry, this was a bug that must have snuck in during refactoring. Thank you for reporting. I just pushed a fix. Can you test Ulauncher from the v6 branch?

**Reply by hedgieinsocks** [reply]
No worries, thanks for the quick fix! I confirm that it works now after the latest commit :+1:
